### PR TITLE
Change Add location to Nearby location if location exists.

### LINKED ISSUE #8566: (multiple issues) Fix UI flow and debug dashboard location input form
We have a feature where you can add your location from the dashboard -- https://publiclab.org/dashboard (while logged in):

image

You click "Add your location" and a modal popup appears, prompting you:

image

Unfortunately, as @Tlazypanda noted, this doesn't really work smoothly, both due to possible UI design gaps (no notification to the user when tags are created) and possibly bugs. @Tlazypanda noted:

> * clicking "Save" doesn't apparently do anything, because you're just dropped back on the dashboard
> * On adding location modal, the slider works by zooming in but when we save it and check it out in the profile page it is saved without any zoom.
> * Also after adding your location it still shows add location on dashboard
> * This is invoked by adding location from dashboard but on clicking share location from profile page nothing happens.
> * Also after adding location for first time, it doesn't seem to update for a second time

Even after fixing this, we also ought to add a system test for adding profile tags - like, one of these modified for the dashboard!

https://github.com/publiclab/plots2/blob/86776c2c72a505b0db563f52d1a8165c36c6eb37/test/system/post_test.rb#L188-L211

We could add one of these assertions afterwards that checks that the profile tags are properly created: 

https://github.com/publiclab/plots2/blob/86776c2c72a505b0db563f52d1a8165c36c6eb37/test/system/tag_test.rb#L40-L51

### Intended workflow

The ideal intended workflow is:

1. See "add a location" on dashboard, click it
2. See modal and check content of one or more of the inputs
3. enter lat/lon and zoom, or use the slider
4. press `Save` and see an alert (maybe?) that says "profile location was saved. Learn more about location privacy" - linking to https://publiclab.org/location-privacy
5. confirm dashboard button now says "View nearby content
6. confirm tags were added and appear on profile page

### PR DESCRIPTION
**Pull request by gauravahlawat81**
Fixes #8566 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

Screencast-from-11-01-21-070256

### CODE REVIEW

File: app/views/tag/_location.html.erb [hunk 1]
```diff
@@ -15,7 +15,7 @@
       <% end %>
       if( $(this).attr("submit_to") ) {
         modalUrl += firstVar ? '?' : '&';
-        modalUrl += 'url=' + encodeURIComponent(locationInput.attr("submit_to"));
+        modalUrl += 'url=' + encodeURIComponent($(this).attr("submit_to"));
```
Comment (sagarpreet-chadha) [inline app/views/tag/_location.html.erb:18]: Why is this change required?

Comment (gauravahlawat81) [reply app/views/tag/_location.html.erb:18]: I don't know why, but this doesn't seem to work, when we select using `locationInput` it returns `undefined`. So I though if we are searching if `submit_to` exists or not using `$(this)`, we can also use the value in same way,

Comment (gauravahlawat81) [reply app/views/tag/_location.html.erb:18]: Also, this change was made for the [PR](https://github.com/publiclab/plots2/pull/8988).

File: app/views/sidebar/_dashboard.html.erb [hunk 2]
```diff
@@ -18,13 +18,20 @@
...
+    <div style="margin: 2.2rem 0;">
+      <a class="btn btn-lg btn-outline-secondary" href="https://publiclab.org/map/#<%= current_user.get_value_of_power_tag("zoom").to_i%>/<%=current_user.get_value_of_power_tag("lat").to_f %>/<%=current_user.get_value_of_power_tag("lon").to_f %>">
+      <i class="fa fa-map-marker" style="color:#c40;"></i> Nearby Activity</a>
+    </div>
+  <%else %>
+    <div style="margin: 2.2rem 0;">
+      <a class="btn btn-lg btn-outline-secondary blurred-location-input" submit_to="/profile/tags/create/<%= current_user.uid %>"><i class="fa fa-map-marker" style="color:#c40;"></i> Add your location</a>
+      <p style="font-size: smaller; margin: 0.3rem 0;">
+        Add your location to see work near you<br>
+        <a href="https://publiclab.org/location-privacy" target="_blank">About location privacy »</a>
```
Comment (Sagarpreet) [inline app/views/sidebar/_dashboard.html.erb:31]: `target="_blank"` will open this in a new window or tab, right? 
@jywarren this is expected behaviour?

**Review by Sagarpreet — APPROVED** [review]
🎉

File: app/views/sidebar/_dashboard.html.erb (hunk 2 above)
Comment (gauravahlawat81) [reply app/views/sidebar/_dashboard.html.erb:31]: It should, but it doesn't in my local. When I click on `Nearby Activity` it doesn't open up a new tab, rather it continues in the same tab. I don't know what the expected behaviour should be :disappointed_relieved: , maybe @jywarren can tell us more?

File: app/views/locations/_form.html.erb [hunk 3]
```diff
@@ -190,6 +190,7 @@
     <% end %>
     
     <% if params[:reload] %>setTimeout(function() { location.reload(); }, 3000);<% end %>
+    notyNotification('relax', 3000, 'success', 'topCenter', 'Location Saved');
```
Comment (RuthNjeri) [inline app/views/locations/_form.html.erb:193]: Hi @gauravahlawat81 I am not sure why this PR is not showing conflicting changes on this file, there were new changes merged to this file a few weeks ago..

https://github.com/publiclab/plots2/blob/f87860ca517d79c487570f871929a452d8ee8bce/app/views/locations/_form.html.erb#L193-L200

Comment (gauravahlawat81) [reply app/views/locations/_form.html.erb:193]: Yup @RuthNjeri , this PR is quite old and I haven't rebased the new code on this branch. I think I will make some more changes and push the updated code again. Thanks for reminding me! :confetti_ball:

**Comment by gauravahlawat81** [comment]
There seems to be some issue with rebasing and pushing the updated code. I will be opening a new PR, sorry about this.
